libraries/AP_Generator/AP_Generator_Cortex.cpp

```diff
@@ -0,0 +1,299 @@
+        return false;
+    }
+
+    if (!is_ready()) {
+        snprintf(failmsg, failmsg_len, "Generator is not ready");
+        return false;
+    }
+
+    return true;
+}
+
+
+void AP_Generator_Cortex::send_generator_status(const GCS_MAVLINK &channel)
+{
+    uint64_t status_flags = 0;
+
+    if (!is_connected()) {
+        return;
+    }
+
+    // Copy status flags from MAV_GENERATOR_STATUS_FLAG enum
+    const Cortex_StatusBits_t &status = telemetry.status.status;
+    const Cortex_WarningBits_t &warning = telemetry.status.warning;
+    const Cortex_ErrorBits_t &error = telemetry.status.error;
+
+    // Check inihibit state
+    if (status.inhibited) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_START_INHIBITED;
+    }
+    
+    switch (status.mode) {
+        case CORTEX_MODE_STANDBY:
+            status_flags |= MAV_GENERATOR_STATUS_FLAG_IDLE;
+            if (status.readyToRun) {
+                status_flags |= MAV_GENERATOR_STATUS_FLAG_READY;
+            }
+            break;
+        case CORTEX_MODE_RUNNING:
+                if (telemetry.generator.current <= 0.05f) {
+                    status_flags |= MAV_GENERATOR_STATUS_FLAG_GENERATING;
+                }
+            break;
+        default:
+            break;
+    }
+
+    // Status flags
+    if (status.readyToRun) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_READY;
+    }
+
+    if (telemetry.battery.current <= 0.1f) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_CHARGING;
+    }
+
+    if (status.powerLimit) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_REDUCED_POWER;
+    }
+
+    // Warning flags
+    if (warning.generatorTempLimit) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_OVERTEMP_WARNING;
+    }
+
+    if (warning.temperature) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_ELECTRONICS_OVERTEMP_WARNING;
+    }
+
+    if (warning.enginePowerLimit) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_MAXPOWER;
+    }
+    
+    // Error flags
+    if (error.generator) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_ELECTRONICS_FAULT;
+    }
+
+    if (error.avionicsRail ||
+        error.payloadRail ||
+        error.servoRail ||
+        error.batteryChargerRail) {
+        status_flags |= MAV_GENERATOR_STATUS_FLAG_POWER_RAIL_FAULT;
+    }
+
+    mavlink_msg_generator_status_send(
+        channel.get_chan(),
+        status_flags,
+        (uint16_t) abs(telemetry.generator.rpm),
+        batteryCurrent(),
+        loadCurrent(),
+        generatorPower(),
+        generatorVoltage(),
+        rectifierTemperature(),
+        std::numeric_limits<double>::quiet_NaN(),
+        telemetry.generator.temperature,
+        telemetry.controller.runTime,
+        0  // time until maintenance (not supported)
+    );
+}
+
+
+bool AP_Generator_Cortex::is_connected(void) const
+{
+    const uint32_t now = AP_HAL::millis();
+
+    return (now - last_reading_ms) < 2000;
```

Comment:
2 seconds seems *way* too generous!

Reply:
Is there a standard or default timeout value to be used here? Otherwise, what would you suggest I reduce it to?

Reply:
No - how long do you think it reasonable that the generator just simply disappear for? :-)

I'm responsible for the existing timeouts being as long as they are - and for the life of me I don't understand how 2 seconds is reasonable on the RichenPower, for example.